letsencrypt/client/tests/client_test.py

```diff
@@ -0,0 +1,88 @@
+"""letsencrypt.client.client.py tests."""
+import unittest
+
+import mock
+import zope.component
+
+
+class RollbackTest(unittest.TestCase):
+    """Test the rollback function."""
+    def setUp(self):
+        self.m_install = mock.MagicMock()
+        self.m_input = mock.MagicMock()
+        zope.component.getUtility = self.m_input
+
+    def _call(self, checkpoints):  # pylint: disable=no-self-use
+        from letsencrypt.client.client import rollback
+        rollback(checkpoints)
+
+    @mock.patch("letsencrypt.client.client.determine_installer")
+    def test_no_problems(self, mock_det):
+        mock_det.side_effect = self.m_install
+
+        self._call(1)
+
+        self.assertEqual(self.m_install().rollback_checkpoints.call_count, 1)
+        self.assertEqual(self.m_install().restart.call_count, 1)
+
+    @mock.patch("letsencrypt.client.reverter.Reverter")
+    @mock.patch("letsencrypt.client.client.determine_installer")
+    def test_misconfiguration_fixed(self, mock_det, mock_rev):
+        from letsencrypt.client.errors import LetsEncryptMisconfigurationError
```

Comment:
This should be imported at the top as `from letsencrypt.client import errors`. I can see there are a lot of similar issues, so I will let you hunt them down.